src/main/java/com/product/dao/DefaultProductDao.java

```diff
@@ -0,0 +1,9 @@
+package com.product.dao;
```

Comment:
this file is actually not needed. I know what might have inspired this. Lets chat offline.

There should just be 1 interface (ProductDao) class not two (DefaultProductDao, ProductDao)

and the implementation class `ProductDaoImpl` should implement `ProductDao` just like you did in CustomerService. 

And for getting the customer, that shoudl be part of another CustomerDao and CustomerDaoImpl, it does not belong to `ProductDao` .. ProductDao should only deal with getting, adding, removing, updating Products.

CustomerDao should be dealing with just Customer.

Reply:
As MongoRepository is Interface that's why i opted this option.